### PR TITLE
dont use cudnn batchnorm for cudnn < 5.1.10

### LINKED ISSUE #1188: cudnn < 6 needs to disable batchnorm codepath
As Arthur Szlam reports, fb-internal cudnn is still lagging behind, and giving batch-size > 1024 with batchnorm is raising an error from cudnn. Need to check for compile-time version and disable this codepath

### PR DESCRIPTION
**Pull request by soumith**
fixes #1188

### CODE REVIEW

File: torch/nn/_functions/batchnorm.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ def forward(self, input, weight=None, bias=None):
         # don't use cuDNN for half inputs because cuDNN requires the weight and
         # bias tensors to be floats, unlike THCUNN which requires half tensors.
         self.use_cudnn = (cudnn.is_acceptable(input) and
+                          cudnn.version() > 5110 and
                           weight is not None and bias is not None and
                           not isinstance(input, torch.cuda.HalfTensor))
```
Comment (apaszke) [inline torch/nn/_functions/batchnorm.py]: Huh, it seems weird that `BatchNormBackward` checks the `training` flag, but the Python function doesn't. Is there any problem with using cuDNN at inference?

Also, we should probably get rid of this and only use the C one.

Comment (soumith) [inline torch/nn/_functions/batchnorm.py]: we cannot use cudnn for backward in inference mode, as it computes incorrectly the gradInput in inference mode

Comment (apaszke) [inline torch/nn/_functions/batchnorm.py]: but is it ok to use cuDNN for forward and THCUNN for backward?

Comment (soumith) [inline torch/nn/_functions/batchnorm.py]: oh hmm, sam says probably not. i'll make them consistent.

Comment (ngimel) [inline torch/nn/_functions/batchnorm.py]: Why? THCUNN with affine transformation and cuDNN should be fully compatible. @eladhoffer had PR #901 that would make batchNorm for inference unnecessary, everyone should use that, it's faster.
